File: android/src/main/kotlin/com/example/pauza_screen_time/usage_stats/model/UsageStatsInterval.kt
package com.example.pauza_screen_time.usage_stats.model

import android.app.usage.UsageStatsManager

/**
 * Typed wrapper for the Android UsageStatsManager interval constants.
 *
 * Use this enum instead of passing raw integers to avoid confusion between
 * arbitrary integers and the valid [UsageStatsManager.INTERVAL_*] constants.
 */
enum class UsageStatsInterval(val rawValue: Int) {
    /** Let the system choose the most appropriate interval. */
    BEST(UsageStatsManager.INTERVAL_BEST),

    /** Aggregate data by day. */
    DAILY(UsageStatsManager.INTERVAL_DAILY),

    /** Aggregate data by week. */
    WEEKLY(UsageStatsManager.INTERVAL_WEEKLY),

    /** Aggregate data by month. */
    MONTHLY(UsageStatsManager.INTERVAL_MONTHLY),

    /** Aggregate data by year. */
    YEARLY(UsageStatsManager.INTERVAL_YEARLY);

    companion object {
        /**
         * Returns the [UsageStatsInterval] matching [rawValue], or [BEST] if not recognised.
         */
        fun fromRawValue(raw: Int): UsageStatsInterval =
            entries.firstOrNull { it.rawValue == raw } ?: BEST
    }
}

File: lib/src/features/usage_stats/model/usage_stats_interval.dart
/// Typed representation of Android's UsageStatsManager interval constants.
///
/// Use this enum when calling [UsageStatsManager.getEventStats] to avoid
/// passing magic integers whose meaning is only documented in comments.
///
/// Values correspond 1-to-1 with the Android `UsageStatsManager.INTERVAL_*`
/// and `UsageStatsManager.INTERVAL_BEST` constants.
///
/// **Android only.**
enum UsageStatsInterval {
  /// Let the system pick the most appropriate interval for the time range.
  best(0),

  /// Aggregate data by day.
  daily(1),

  /// Aggregate data by week.
  weekly(2),

  /// Aggregate data by month.
  monthly(3),

  /// Aggregate data by year.
  yearly(4);

  /// The raw integer constant passed to the Android platform.
  final int rawValue;

  const UsageStatsInterval(this.rawValue);

  /// Returns the [UsageStatsInterval] for [rawValue], or [best] if not recognised.
  static UsageStatsInterval fromRawValue(int value) =>
      values.firstWhere((e) => e.rawValue == value, orElse: () => best);
}
